src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -281,29 +353,38 @@ void func_80AF2AB4(EnRu2* this, PlayState* play) {
     }
 }
 
-void func_80AF2B44(EnRu2* this, PlayState* play) {
+/**
+ * Checks to see if the blue warp should be spawned for Ruto to rise up through.
+ */
+void EnRu2_CheckBlueWarp(EnRu2* this, PlayState* play) {
     CutsceneContext* csCtx = &play->csCtx;
     CsCmdActorCue* cue;
 
     if (csCtx->state != CS_STATE_IDLE) {
         cue = csCtx->actorCues[3];
 
         if ((cue != NULL) && (cue->id == 2)) {
-            this->action = 2;
-            this->drawConfig = 1;
-            func_80AF29DC(this, play);
+            this->action = ENRU2_RISE_THROUGH_BLUE_WARP;
+            this->drawConfig = ENRU2_DRAW_OPA;
+            EnRu2_SpawnBlueWarp(this, play);
         }
     }
 }
 
-void func_80AF2B94(EnRu2* this) {
+/**
+ * Halts Ruto's rise up through the blue warp in the Chamber of Sages once finished.
+ */
+void EnRu2_EndRise(EnRu2* this) {
     if (this->actor.shape.yOffset >= 0.0f) {
-        this->action = 3;
+        this->action = ENRU2_SAGE_OF_WATER_DIALOG;
         this->actor.shape.yOffset = 0.0f;
     }
 }
 
-void func_80AF2BC0(EnRu2* this, PlayState* play) {
+/**
+ * Sets up the animation for Ruto to raise her arms to give Link the Water Medallion.
+ */
+void EnRu2_CueRaiseArmsAnimation(EnRu2* this, PlayState* play) {
```

Comment:
This should probably be named after "check" like `CheckToSpawnBlueWarp`
`CheckToRaiseArms` seems fine. Dont think cue needs to be mentioned in this context, its a given that all this cutscene stuff is controlled with cues